clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -92,6 +93,13 @@ public MetadataRequest build(short version) {
             if (!data.allowAutoTopicCreation() && version < 4)
                 throw new UnsupportedVersionException("MetadataRequest versions older than 4 don't support the " +
                         "allowAutoTopicCreation field");
+            if (data.topics() != null) {
```

Comment:
Can we remove the unnecessary `()` from the various files?

Reply:
I think this one is needed as we can't remove in java files. I tried to remove all of the ones in scala files.